### PR TITLE
Tidy ups

### PR DESCRIPTION
**Pull request by ltratt**
The first commit is stuff suggested by Clippy (the `write_all` aspect is actually important!). That then made me realise that I could use the `write!` macro (why this eluded my attention the first time around is an open question!).

### CODE REVIEW

File: src/lib/parser.rs [hunk 1]
```diff
@@ -128,12 +128,9 @@ impl<TokId: TryFrom<usize>> LexParser<TokId> {
...
             name = None;
         } else {
-            debug_assert!(orig_name.len() > 0);
-            if !((orig_name.chars().nth(0).unwrap() == '\''
-                  && orig_name.chars().last().unwrap() == '\'')
-                 ||
-                 (orig_name.chars().nth(0).unwrap() == '\"'
-                  && orig_name.chars().last().unwrap() == '"')) {
+            debug_assert!(!orig_name.is_empty());
+            if !(   (orig_name.starts_with('\'') && orig_name.ends_with('\''))
```
Comment (snim2) [inline src/lib/parser.rs:132]: Why the spaces between the parens?

**Comment by snim2** [comment]
There is also `writeln!` I think. Just one minor comment, otherwise this looks good to me.

File: src/lib/parser.rs (hunk 1 above)
Comment (ltratt) [reply src/lib/parser.rs:132]: So the `||` lines up nicely.

Comment (snim2) [reply src/lib/parser.rs:132]: OK, first time I've seen something like this, but I'll believe you!

**Review by snim2 — APPROVED** [review]

**Comment by snim2** [comment]
Do you need to squash before I merge?

**Comment by ltratt** [comment]
I think the commits are best left separate. Ready to merge!

**Pull request merged**